test_util/marathon.py

```diff
@@ -14,91 +17,140 @@
 log = logging.getLogger(__name__)
 
 
-def get_test_app(custom_port=False):
+class Container(enum.Enum):
+    DOCKER = 'DOCKER'
+    MESOS = 'MESOS'
+    NONE = None
+
+
+class Network(enum.Enum):
+    HOST = 'HOST'
+    USER = 'USER'
+    BRIDGE = 'BRIDGE'
+
+
+class Healthcheck(enum.Enum):
+    HTTP = 'HTTP'
+    MESOS_HTTP = 'MESOS_HTTP'
+
+
+def get_test_app(
+        host_port: int=0,
+        container_port: int=None,
+        container_type: Container=Container.NONE,
+        network: Network=Network.HOST,
+        healthcheck_protocol: Healthcheck=Healthcheck.HTTP,
+        vip: str=None,
+        host_constraint: str=None):
+    """ Creates an app definition for the python test server which will be
+    consistent (i.e. deployable with green health checks and desired network
+    routability). To learn more about the test server, see in this repo:
+    ../packages/dcos-integration-test/extra/util/python_test_server.py
+
+    Args:
+        host_port: port that marathon will use to route traffic into the
+            test server container. If set to zero, then marathon will assign
+            a port (which is referenced by index).
+        container_port: if network is BRIDGE, then the container can have a
+            port remapped inside the container. In HOST or USER network, the
+            container port must be the same as the host port
+        container_type: can be NONE (default Mesos runtime), MESOS (the UCR),
+            or DOCKER
+        health_check_protocol: can be MESOS_HTTP or HTTP
+        vip: either named or unnamed VIP to be applied to the host port
+        host_constraint: string representing a hostname for an agent that this
+            app should run on
+
+    Return:
+        (dict, str): 2-Tuple of app definition (dict) and app ID (string)
+    """
+    if network == Network.BRIDGE:
+        assert container_type == Container.DOCKER, 'BRIDGE network mode only supported for DOCKER container type'
```

Comment:
Yes. Absolutely. assert is intended to stop programmer errors last I checked and this is a great example; if a user wants a Network.BRIDGE app, then it MUST be DOCKER. Nothing else is valid or sensible. If DOCKER is not being used, then the programmer must not be knowledgeable about what they are doing or what this test application actually is.